quickcheck/Test/Tasty/QuickCheck.hs

```diff
@@ -283,8 +283,31 @@ quickCheck yieldProgress args
   = (.) (QC.quickCheckWithResult args)
   $ QCP.callback
   $ QCP.PostTest QCP.NotCounterexample
-  $ \QC.MkState {QC.maxSuccessTests, QC.numSuccessTests} _ ->
-    yieldProgress $ emptyProgress {progressPercent = fromIntegral numSuccessTests / fromIntegral maxSuccessTests}
+  $ \st@QC.MkState {QC.maxSuccessTests, QC.numSuccessTests} _ ->
+    yieldProgress $
+      if QC.numTotTryShrinks st > 0 then
+        emptyProgress {
+            progressText = showShrinkCount st True
+          }
+      else
+        emptyProgress {
+            progressPercent = fromIntegral numSuccessTests / fromIntegral maxSuccessTests
+          }
+
+-- Based on 'QuickCheck.Test.failureSummaryAndReason'.
+showShrinkCount :: QC.State -> Bool -> String
+showShrinkCount st full = count
+  where
+    count :: String
+    count =
+      concat [
+        show (QC.numSuccessShrinks st) ++
+        concat [ "." ++ show (QC.numTryShrinks st) | showNumTryShrinks ] ++
```

Comment:
I spent quite some time deciphering that `concat [ xs | c ]` is a fancy way to say `if c then xs else ""`. Could we please use the latter form?

Reply:
The intermediate output seems to be something like `2.1 shrinks`. What is it supposed to mean? 

I understand that you adapted `QuickCheck.Test.failureSummaryAndReason`, which has the same cryptic behavior. Could we just `show (QC.numSuccessShrinks st) ++ " shrink" ++ (if ... then "" else "s")`?